### ISSUE TITLE
useSliderThumb - excludeFromTabOrder won't modify the tabIndex

### ISSUE THREAD

**Issue by Persists** [adobe/react-spectrum#6727]
### Provide a general summary of the issue here

The `excludeFromTabOrder` prop on the `sliderThumb` component in `react-aria-components` does not work as intended. Despite setting the prop to `true`, the component remains in the tab order during keyboard navigation.

### 🤔 Expected Behavior?

When the `excludeFromTabOrder` prop is set to `true` on the `sliderThumb` component in `react-aria-components`, the component should be removed from the tab order.

### 😯 Current Behavior

The `sliderThumb` component in` react-aria-components` supports the `excludeFromTabOrder` prop, but this prop does not function as expected. The component is not excluded from the tab order when set to `true`.

### 💁 Possible Solution

In the `useSliderThumb` in `packages/@react-aria/slider/src/useSliderThumb.ts` overwrites the tabIndex in the return statement. Maybe it could be done like in the `useButton` Hook, where the tabIndex is calculated before the return. 

### 🖥️ Steps to Reproduce

[Link to Codesandbox, press `tab` and see whats happening](https://codesandbox.io/p/sandbox/condescending-blackburn-9552fy?file=%2Fsrc%2FApp.js%3A14%2C3)

### Version

latest

### What browsers are you seeing the problem on?

Firefox

### If other, please specify.

All

### What operating system are you using?

MacOs

**Comment by snowystinger** [comment]
Can you explain what you need that prop for? Examples would be helpful

**Comment by Persists** [comment]
Firstly, the prop is included in the `SliderThumb` props, so I assumed it works as intended.

Example: I have a form with several input fields and a slider. The slider is only relevant based on previous input, and I want to exclude it from the tab order if the input is not set, 

In another scenario, also in a form, I might want users to navigate only through input fields and skip sliders.

**Comment by snowystinger** [comment]
It is regrettable that the prop ended up appearing in the types, however, we did not intend to expose it there. Thank you for bringing it to our attention. I will be removing it as it wasn't doing anything.

Thanks for the extra info about your use case, however, excludeFromTabOrder won't help you with those cases. It only removes the element from keyboard navigation, it'd still be functional to a touch, mouse, or Assistive technology user.

Instead of excluding the slider from the tab order, I have a few other suggestions.
- use disabled or readonly
- wrap the slider in an inert element
- do not render the slider

**Comment by Persists** [comment]
Thank you, that helped me a lot!
